src/github.com/getlantern/flashlight/server/server.go

```diff
@@ -222,6 +250,69 @@ func (server *Server) stopNattywad() {
 	server.waddellClient = nil
 }
 
+func (server *Server) checkForDisallowedPort(addr string) error {
+	_, portString, err := net.SplitHostPort(addr)
+	if err != nil {
+		return fmt.Errorf("Unable to split host and port for %v: %v", addr, err)
+	}
+
+	port, err := strconv.Atoi(portString)
+	if err != nil {
+		return fmt.Errorf("Unable to convert port %v to integer: %v", portString, err)
+	}
+
+	portAllowed := false
+	for _, allowed := range server.AllowedPorts {
+		if port == allowed {
+			portAllowed = true
+			break
+		}
+	}
+
+	if !portAllowed {
+		return fmt.Errorf("Not accepting connections to port %v", port)
+	}
+
+	return nil
+}
+
+func (server *Server) checkForDisallowedCountry(req *http.Request) error {
+	clientIp := getClientIp(req)
+	if clientIp == "" {
+		log.Debug("Unable to determine client ip for geolookup")
+		return nil
+	}
+
+	country := ""
+	cachedCountry, found := server.geoCache.Get(clientIp)
+	if found {
+		log.Tracef("Country for %v found in cache", clientIp)
+		country = cachedCountry.(string)
+	} else {
+		log.Tracef("Country for %v not cached, perform geolookup", clientIp)
+		city, err := geolookup.LookupCity(clientIp)
+		if err != nil {
```

Comment:
An error in this case means a network error calling the geoip service.  If the geoip service returns an unknown country, that's not an error, just a blank country.